### DISCUSSION TITLE
Prevent showing an instantiated scenes Child Nodes in the Group Editor unless "Editable Children" is enabled

### DISCUSSION THREAD

**Discussion by FelixSeKett** [godotengine/godot-proposals#8631] [Editor]
Hello everyone,

the Group Editor shouldn't show the child nodes of an instantiated scene for adding and removing them to and from groups by default. It implies that those child nodes could be managed this way, but thats not the case - until "Editable Children" is checked for the respective scene in the scene tree. According to the current behaviour of the group editor, such an added child node would be part of a group, but during runtime it is not. Therefore, they should only be shown in the editor after one marked the checkbox of their respective scene instance. Especially for newcomers, who are not used to the ways of Godot, the current handling can be quite confusing.

Here is an example: The instantiated Hero scene has no "Editable Children". Therefore its Sprite is not visible in the scene tree. Until one checks this checkbox it shouldn't be visible in the Group Editor either, for it can't be managed runtime-consistent anyway.

<img width="224" alt="Screenshot 2023-12-04 at 12 31 53" src="https://github.com/godotengine/godot-proposals/assets/20595908/32919dc8-6b62-4448-8cec-c0b6ef06e003">
<img width="226" alt="Screenshot 2023-12-04 at 12 37 30" src="https://github.com/godotengine/godot-proposals/assets/20595908/0a53a0fd-f522-45f4-8042-20531f9452ec">
<img width="1038" alt="Screenshot 2023-12-04 at 12 35 09" src="https://github.com/godotengine/godot-proposals/assets/20595908/b2499d18-3759-4024-b240-3bc7af002c36">

I am not very experienced with the engine, but people who seem to be told me that this would indeed be quite an illogical behaviour and I should propose an improvement. So let me know what you think!

**Comment by Calinou** [comment]
This sounds like a bug, especially since you're effectively setting data that is lost when the editor quits (which is why the running project ignores it).

**Reply by FelixSeKett** [reply]
Yeah, thats what the others said. I simply couldn't imagine no one inside Godot noticed this before. In the end, they are encouraging people to not enable "Editable Children" by having this option disabled by default. The possibility that obviously no one working this way had ever noticed this bug before is hard to believe. I will wait a few more days for some additional comments to come until I report the bug. Maybe there is an intention behind this only a few chosen ones know.

**Reply by Calinou** [reply]
> The possibility that obviously no one working this way had ever noticed this bug before is hard to believe.

We don't have an extensive testing process with paid QA, so we sometimes miss things that may be obvious to some :slightly_smiling_face: 

[Testing pull requests](https://docs.godotengine.org/en/latest/contributing/workflow/testing_pull_requests.html) before they are merged is something we encourage everyone to do, but it's surprisingly difficult to get people to do it. We've tried increasing our efforts to make PRs that need testing more visible (e.g. with a dedicated Discord channel), but there's only so much we can do.

**Reply by AThousandShips** [reply]
If you want to help with minimizing bugs at release you are very welcome to test yourself 🙂

**Reply by FelixSeKett** [reply]
I am terribly sorry, I didn't want to offend anyone! 🙏🏽 I am very thankful for your incredible work with the engine and simply didn't know how the whole releasing and testing process works. I am still new to the whole engine world, not to mention the open source world, and obviously had the wrong idea about it. For now, I will do the only part I am capable of by upgrading this discussion to an actual issue so you guys can keep this topic on your radar. 🙂
